crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -121,6 +121,19 @@ def checks
             upgrade_status.end_step
           end
         end
+      rescue ::Crowbar::Error::StartStepRunningError,
+             ::Crowbar::Error::StartStepOrderError,
+             ::Crowbar::Error::SaveUpgradeStatusError => e
+        raise ::Crowbar::Error::UpgradeError.new(e.message)
+      rescue StandardError => e
+        ::Crowbar::UpgradeStatus.new.end_step(
```

Comment:
good point, but i guess it doesn't need to be in all the steps, as they are always running implicitly once you hit this exception. so i guess your suggestion applies for the prechecks here.